### PR TITLE
[FLINK-36206][Connectors/AWS] Adding support for custom override configuration

### PR DESCRIPTION
**Pull request by gguptp**
## Purpose of the change

Adding support for custom override configuration in AWSClientUtil.java

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change added tests and can be verified as follows:

- *Added unit tests*

## Significant changes
*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterwards, for convenience.)*
- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [ ] New feature has been introduced
  - If yes, how is this documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java [hunk 1]
```diff
@@ -207,6 +207,39 @@ S createAwsSyncClient(
                 .build();
     }
 
+    public static <
```
Comment (hlteoh37) [inline flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java:210]: can we collapse the previous `createAwsSyncClient()` into this one?

**Review by hlteoh37 — CHANGES_REQUESTED** [review]
Thanks @gguptp for the contribution! Added 1 comment.

File: flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java (hunk 1 above)
Comment (gguptp) [reply flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java:210]: sure

File: flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java [hunk 2]
```diff
@@ -194,7 +196,7 @@ S createAwsSyncClient(
         final ClientOverrideConfiguration overrideConfiguration =
                 createClientOverrideConfiguration(
                         clientConfiguration,
-                        ClientOverrideConfiguration.builder(),
+                        clientOverrideConfigurationBuilder,
```
Comment (hlteoh37) [inline flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java:199]: Let's keep the old method to keep backwards compatibility

Comment (gguptp) [reply flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java:199]: my bad, fixed this in next revision

**Pull request merged**